### PR TITLE
Used newspaper

### PR DESCRIPTION
**Pull request by mphirke**
I read Slack and implemented "- Using Newspaper instead of BeautifulSoup [Provides all news scraping capabilities]"

### CODE REVIEW

**Comment by stephenh67** [comment]
the bs4 import is still included

**Comment by asiffarhankhan** [comment]
Beautiful soup library does not have any functionality. I will merge this and delete that line on the next commit

File: modules/headline.py [hunk 1]
```diff
@@ -2,6 +2,8 @@
 from bs4 import BeautifulSoup
```
Comment (mandjevant) [inline modules/headline.py]: bs4 not necessary.

File: modules/headline.py [hunk 2]
```diff
@@ -2,6 +2,8 @@
...
 from textblob.classifiers import NaiveBayesClassifier
```
Comment (mandjevant) [inline modules/headline.py:2]: Textblob not necessary.

File: modules/headline.py [hunk 3]
```diff
@@ -12,16 +14,18 @@ def __init__(self):
 
     def extract_headline(self):
         self.net_con=True #Expecting Internet Connection to be working initially
```
Comment (mandjevant) [inline modules/headline.py]: not necessary

File: modules/headline.py [hunk 4]
```diff
@@ -12,16 +14,18 @@ def __init__(self):
...
+        
         try:
```
Comment (mandjevant) [inline modules/headline.py:12]: No need for a try except

**Review by mandjevant — CHANGES_REQUESTED** [review]

**Comment by mandjevant** [comment]
> 
> 
> Beautiful soup library does not have any functionality. I will merge this and delete that line on the next commit

Nope, the one making the pull request should supply good code with no redundancy and good efficiency. This is to minimize the actions to master directly. Actions directly to master should not be done. Ever.

File: modules/headline.py (hunk 1 above)
Comment (mphirke) [reply modules/headline.py]: Will remove.

File: modules/headline.py (hunk 2 above)
Comment (mphirke) [reply modules/headline.py:2]: Required in the train_data and the headline_category as of now. Can be removed once NLTK is implemented.

File: modules/headline.py (hunk 3 above)
Comment (mphirke) [reply modules/headline.py]: It was a part of the original code, but removed.

File: modules/headline.py (hunk 4 above)
Comment (mphirke) [reply modules/headline.py:12]: Removed, although try and except seems like a good practice to me.

Comment (asiffarhankhan) [reply modules/headline.py:12]: Hey there is a need for try except I believe. When internet is not connected it gives a hell of a long error messege. Also hampers other functions task

**Comment by asiffarhankhan** [comment]
@mandjevant sure

File: modules/headline.py (hunk 2 above)
Comment (asiffarhankhan) [reply modules/headline.py:2]: I am working on it. learning how to train data using the same classifier with NLTK. Once done I will make a pull request

File: modules/headline.py [hunk 5]
```diff
@@ -1,28 +1,18 @@
...
     def extract_headline(self):
-        self.net_con=True #Expecting Internet Connection to be working initially
-        try:
-            news_page=urllib.request.urlopen(self.news_url)   
-            soup = BeautifulSoup(news_page,'html.parser')
-            headline_in_html=soup.find('h1')
-            headline=headline_in_html.text.strip()
-            return headline
-
-        except urllib.error.URLError:
```
Comment (asiffarhankhan) [inline modules/headline.py:22]: I am waiting for @mandjevant reply. I have requested for this block to be preserved. as it gives a page of error when there is no internet. Hence the variable "net_con" was required too

File: modules/headline.py [hunk 6]
```diff
@@ -1,28 +1,18 @@
...
-            print("\nCONNECTIION ERROR:There may be a connection problem. Please check if the device is connected to the Internet")
-            self.net_con=False #Value update if the program is unable to connenct
-
+        article = newspaper.Article(self.news_url)
```
Comment (asiffarhankhan) [inline modules/headline.py:21]: Using Newspaper for scraping headline seems like a better switch

**Review by asiffarhankhan — CHANGES_REQUESTED** [review]

File: modules/headline.py (hunk 5 above)
Comment (mandjevant) [reply modules/headline.py:22]: The block wont work when Urllib isnt requesting a site. (Since we are using newspaper now)

Comment (mandjevant) [reply modules/headline.py:22]: Additionally, you should make a seperate statement at the begining of the file that returns an error and sys.exit()'s the program.

File: modules/headline.py (hunk 4 above)
Comment (mandjevant) [reply modules/headline.py:12]: Downloading and parsing the article cant go wrong when there is an internet connection. Testing for internet connection should be done in __init__. So here, the try-except is not necessary.

File: modules/headline.py (hunk 5 above)
Comment (mphirke) [reply modules/headline.py:22]: So I shouldn't make any changes now, right @mandjevant @asiffarhankhan ?

Comment (mandjevant) [reply modules/headline.py:22]: @mphirke Regarding this, you should remove block and make a network connection check in __init__.

**Comment by mphirke** [comment]
I have made the required changes. Please confirm @mandjevant

**Comment by mandjevant** [comment]
> 
> 
> I have made the required changes. Please confirm @mandjevant

Looks good to me!

**Comment by mphirke** [comment]
Can this be merged?

**Comment by asiffarhankhan** [comment]
@mphirke i dont think that will be a problem. Cuz I have made all the changes on top of your code. I had copied headline.py from your fork and then added the changes. So as long as other files don disrupt we'll be fine

**Review by asiffarhankhan — APPROVED** [review]

**Pull request merged**